src/spikeinterface/extractors/sinapsrecordingh5extractor.py

```diff
@@ -0,0 +1,112 @@
+from pathlib import Path
+import numpy as np
+
+from ..core.core_tools import define_function_from_class
+from ..core import BaseRecording, BaseRecordingSegment
+
+class SinapsResearchPlatformH5RecordingExtractor(BaseRecording):
+    extractor_name = "SinapsResearchPlatformH5"
+    mode = "file"
+    name = "sinaps_research_platform_h5"
+
+    def __init__(self, file_path):
+
+        try:
+            import h5py
+            self.installed = True
+        except ImportError:
+            self.installed = False
+
+        assert self.installed, self.installation_mesg
+        self._file_path = file_path
+
+        sinaps_info = openSiNAPSFile(self._file_path)
+        self._rf = sinaps_info["filehandle"]
+
+        BaseRecording.__init__(
+            self,
+            sampling_frequency=sinaps_info["sampling_frequency"],
+            channel_ids=sinaps_info["channel_ids"],
+            dtype=sinaps_info["dtype"],
+        )
+
+        self.extra_requirements.append("h5py")
+
+        recording_segment = SiNAPSRecordingSegment(
+            self._rf, sinaps_info["num_frames"], sampling_frequency=sinaps_info["sampling_frequency"]
+        )
+        self.add_recording_segment(recording_segment)
+
+        # set gain
+        self.set_channel_gains(sinaps_info["gain"])
+        self.set_channel_offsets(sinaps_info["offset"])
+
+        # set other properties
+
+        self._kwargs = {"file_path": str(Path(file_path).absolute())}
+
+    def __del__(self):
+        self._rf.close()
+
+class SiNAPSRecordingSegment(BaseRecordingSegment):
+    def __init__(self, rf, num_frames, sampling_frequency):
+        BaseRecordingSegment.__init__(self, sampling_frequency=sampling_frequency)
+        self._rf = rf
+        self._num_samples = int(num_frames)
+        self._stream = self._rf.require_group('RealTimeProcessedData')
+
+    def get_num_samples(self):
+        return self._num_samples
+
+    def get_traces(self, start_frame=None, end_frame=None, channel_indices=None):
+        if isinstance(channel_indices, slice):
+            traces = self._stream.get('FilteredData')[channel_indices, start_frame:end_frame].T
+        else:
+            # channel_indices is np.ndarray
+            if np.array(channel_indices).size > 1 and np.any(np.diff(channel_indices) < 0):
+                # get around h5py constraint that it does not allow datasets
+                # to be indexed out of order
+                sorted_channel_indices = np.sort(channel_indices)
+                resorted_indices = np.array([list(sorted_channel_indices).index(ch) for ch in channel_indices])
+                recordings = self._stream.get('FilteredData')[sorted_channel_indices, start_frame:end_frame].T
+                traces = recordings[:, resorted_indices]
+            else:
+                traces = self._stream.get('FilteredData')[channel_indices, start_frame:end_frame].T
+        return traces
```

Comment:
@NinelK if the data is stored and retrieved as `unit16`, then applying the `gain` and `offset` with SI won't work, because it assumes that the data is signed. Have you tried to retrieve traces with `return_scaled=True`?

Reply:
Yes, `return_scaled=True` works. You are looking at the SiNAPSRecordingSegment, which does not scale anything.

SinapsResearchPlatformH5RecordingExtractor is based on BaseRecording, which applies scaling.

Reply:
That's what I tested
```
ch_0, n_ch = 0, 100
h5_traces = h5_recording.get_traces(start_frame=0, end_frame=2000, channel_ids=np.arange(ch_0,ch_0+n_ch), return_scaled=True)
bin_traces = bin_recording.get_traces(start_frame=0, end_frame=2000, channel_ids=np.arange(ch_0,ch_0+n_ch),return_scaled=True)

assert np.all(h5_traces==bin_traces)
```

Without rescaling these outputs are different, and have different types (`uint16` vs `int16`).

Reply:
Ok so probably the H5 converter tranasforms the data to int16. Thanks for checking!

Reply:
It is not converted to `int16`, but scaling still works.
![image](https://github.com/alejoe91/spikeinterface/assets/1615928/dd008c39-4ba4-4ccf-b6af-92c8bf2eea7a)

Reply:
Ah I think we deal with it at the `get_traces` level: https://github.com/SpikeInterface/spikeinterface/blob/main/src/spikeinterface/core/baserecording.py#L252

Reply:
hm... this is not good. Because now if I do `cast_unsigned=True`, then the traces are wrong.
Should I then turn it into `int16` in the extractor?

Reply:
right...we should probably deprecate that!